DO NOT MERGE - Migrate to core24 and gnome-46-2404

bandali opened the pull request:
This PR switches Firefox stable snap from core22 and gnome-42-2204 to core24 and gnome-46-2404.

This **must not be merged** yet, given https://launchpad.net/bugs/2043993 ~~as well as currently missing armhf builds~~, and that we should do the switch for ~~Nightly and~~ Beta first, and give advance heads up about the change on the Ubuntu Community Hub (Discourse) as we did for the switch to core22.

However, in the mean time, builds for this branch are available from the `latest/candidate/core24` channel of the `firefox` snap on the snap store for testing purposes.

Commit:
Migrate to core24 and gnome-46-2404

seb128 commented:
Thanks, any reason to not have started with nightly which we could probably merge earlier?

Commit:
Bug 1921018 - Stage missing libs for libavcodec60

seb128 committed:
Workaround $CRAFT_PARALLEL_BUILD_COUNT issue

seb128 committed:
Cherry-pick 'Allow DRI/EGL path when running under Snap (with core24)'

seb128 committed:
Merge pull request #90 from seb128/stable-core24-hwdrendering

Cherry-pick 'Allow DRI/EGL path when running under Snap (with core24)'

seb128 committed:
Don't try to stage libvpl on !amd64, the package is intel specific

seb128 committed:
Backport an extra commit to fix access to the gpu content

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Bump version to the latest release candidate (131.0.2-1).
  Add layout to opensc-pkcs11.so to make loading it less ugly.
  Stage opensc-pkcs11.so.
  Add PCSCD interface for LP:1967632.

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Update to 131.0.3
  Add instructions on how to report security issues

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Bump version to the current release candidate (132.0-1)
  Git push to the default branch
  Remove the use of a personal token
  Try if changing the schedule impact action owner...
  Update the checkout action to the current version

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Update to 133.0-1
  Merge beta for stable release 133.
  Use the right esr branch in the github action
  Bump version to the latest release candidate (132.0.2-2).
  Bump version to the latest release candidate (132.0.2-1).
  Bump version to the latest release candidate (132.0.1-2).
  Bump version to the latest release candidate (132.0.1-1).
  Check for updates in the current esr serie

seb128 committed:
Remove patches included in the new version

seb128 committed:
Merge branch 'stable' into stable-core24

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Bump version to the latest release candidate (134.0-1).
  Bump version to the latest release candidate (133.0.3-1).
  Add mount-observe plug for https://bugzilla.mozilla.org/1934713
  And don't specify the branch name when pushing to esr...
  Don't specify the branch name when pushing to beta either
  Revert 'git push stable' back to 'git push'.

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Bump version to the latest release candidate (134.0.2-1).
  Also run dictionary installation hook in a first installation of the snap.
  Fix lack of spell checking for a given language if its hunspell file is a symlink.
  Bump version to the latest release candidate (134.0.1-1).

seb128 committed:
Merge branch 'stable' into stable-core24
* stable: (586 commits)
  Bump version to the latest beta (135.0b9-1).
  Bump version to the latest beta (135.0b8-1).
  Bump version to the latest beta (135.0b7-1).
  Bump version to the latest beta (135.0b6-1).
  Bump version to the latest beta (135.0b5-1).
  Bump version to the latest beta (135.0b4-1).
  Bump version to the latest beta (135.0b3-1).
  Set MOZ_APP_REMOTINGNAME to match the desktop file name #74
  Also run dictionary installation hook in a first installation of the snap.
  Fix lack of spell checking for a given language if its hunspell file is a symlink.
  Bump version to the latest beta (135.0b2-1).
  Bug 1661935 - Flip Native Messaging portal from to enable use on Snap
  Bump version to the latest beta (135.0b1-1).
  Add mount-observe plug for https://bugzilla.mozilla.org/1934713
  And don't specify the branch name when pushing to esr...
  Don't specify the branch name when pushing to beta either
  Revert 'git push stable' back to 'git push'.
  Bump version to the latest beta (134.0b10-1).
  Bump version to the latest beta (134.0b9-1).
  Bump version to the latest beta (134.0b8-1).
  ...

seb128 committed:
post-refresh: handle the case of dictionaries not available

https://github.com/ubuntu/app-center/issues/1891

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Bump version to the latest beta (137.0b1-1).
  Bump version to the latest beta (136.0b9-1).
  Bump version to the latest beta (136.0b8-1).
  Bump version to the latest beta (136.0b7-1).
  Bump version to the latest release candidate (135.0.1-1).
  Bump version to the latest beta (136.0b6-1).
  Bump version to the latest beta (136.0b5-1).
  post-refresh: handle the case of dictionaries not available
  post-refresh: handle the case of dictionaries not available
  Bump version to the latest beta (136.0b4-1).
  Bump version to the latest beta (136.0b3-1).
  Bump version to the latest beta (136.0b2-1).
  Bump version to the latest beta (136.0b1-4).
  Bump version to the latest beta (136.0b1-3).
  Bump version to the latest beta (136.0b1-2).
  Bump version to the latest beta (136.0b1-1).

seb128 committed:
Use a non deprecated Ubuntu image for the workflows

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Only send Mattermost notifications once a day.
  Bump version to the latest release candidate (136.0.3-1).
  Bump version to the latest beta (137.0b10-1).
  Bump version to the latest beta (137.0b9-1).
  Bump version to the latest beta (137.0b8-1).
  Bump version to the latest release candidate (136.0.2-1).
  Bump version to the latest beta (137.0b7-1).
  Bump version to the latest beta (137.0b6-1).
  Bump version to the latest beta (137.0b5-1).
  Bump version to the latest release candidate (136.0.1-1).
  Bump version to the latest beta (137.0b4-1).
  Bump version to the latest beta (137.0b3-1).
  Use a non deprecated Ubuntu image for the workflows
  Use a non deprecated Ubuntu image for the workflows
  Bump version to the latest beta (137.0b2-1).
  Send notification in Mattermost when a manual merge is due.
  Bump rust version to 1.82
  Set 'firefox_firefox' as the StartupWMClass hint in the desktop file.

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Bump version to the latest release candidate (137.0.2-1).
  Bump version to the latest release candidate (137.0.1-1).
  snapcraft.yaml: support rust riscv64 download
  Bump version to the latest release candidate (137.0-2).

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  New stable release 138.0-1.

nteodosio committed:
Reduce parallelism of PGO build on Launchpad.

Builds get OOM way too often in Launchpad while compiling gkrust,
so reduce parallelism in Launchpad.

nteodosio committed:
Revert "Reduce parallelism of PGO build on Launchpad."

This reverts commit 78fd7e12716949065cca533d0ce5100c85e665f1.

It turned out that even with 1 job the build would still with the same
frequency exhaust the 16 GB of memory available in Launchpad.

https://github.com/canonical/firefox-snap/pull/130

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Bump version to the latest release candidate (139.0-2).
  Use thin LTO in amd64 too.
  New stable release 139.0-1.
  Bump version to the latest release candidate (138.0.4-1).
  Bump version to the latest release candidate (138.0.3-1).
  Bump version to the latest release candidate (138.0.1-1).

tsteven4 commented:
The content for /usr/share/libdrm has moved from gnome-platform to gpu-2404, thus the layout needs to change from

>  /usr/share/libdrm:
    bind: $SNAP/gnome-platform/usr/share/libdrm

to 
 > /usr/share/libdrm:
     bind: $SNAP/gpu-2404/libdrm

This can be seen searching for amdgpu.ids:
$ find $SNAP -name amdgpu.ids
/snap/firefox/6231/gpu-2404/libdrm/amdgpu.ids

seb128 committed:
Merge branch 'stable' into stable-core24
* stable:
  Bump version to the latest release candidate (139.0.4-1).
  Bump version to the latest release candidate (139.0.1-1).

tsteven4 committed:
correct libdrm location for gnome-46-2404.

tsteven4 commented:
The fix for /usr/share/libdrm is more complicated, see #133, #134.

seb128 committed:
Merge branch 'stable' into stable-core24

tsteven4 committed:
remove unnecessary layout for libdrm.

nteodosio committed:
Merge pull request #134 from tsteven4/stable-libdrm

[stable-core24] correct libdrm location for gnome-46-2404.

lissyx committed:
Bug 259356 - Force legacy behavior for XDG support on Snap (#138)

This is not saying bug 1902632 /
https://github.com/canonical/firefox-snap/pull/63 is wrong but rather
that this requires upstream (gnome-sdk) changes and that because of how
Snapd treats those directories (SNAP_USER_DATA/SNAP_USER_COMMON
https://snapcraft.io/docs/environment-variables#heading--snap-user-common)
there may be risks.

Addressing XDG support for Firefox is not as important when Snap is
involved given the profile is already a subdirectory of the snap user's
data under $HOME/snap/<snap-package>/..., so let's make use of the
legacy behavior until all the upstream risks are well understood.

Co-authored-by: Alexandre <alexandre@serveur>

nteodosio committed:
CI: Ignore major version number for new version check for ESRs.

nteodosio committed:
Merge branch 'stable' into HEAD

nteodosio committed:
Retry downloads of the LLVM tarball.

Github introduced a time-out that causes this pull to fail in ARMv8
Launchpad builders. Use -c to continue a failed pull. Give up after
6 attempts.

nteodosio committed:
Merge branch 'stable' into HEAD

nteodosio committed:
Upstream release 143.0-1.

nteodosio committed:
Go back to CRAFT_PARALLEL_BUILD_COUNT.

https://github.com/canonical/snapcraft/issues/4785 was fixed.

brlin-tw commented:
Hello, are there any blockers currently preventing the change from landing on the stable branch?  If so, do you have any tips on where one can locate them?

nteodosio commented:
Hi, yes: https://launchpad.net/bugs/2043993.

Am 20/09/2025 um 02:23 schrieb ***@***.***:
> brlin-tw left a comment (canonical/firefox-snap#65)
> 
> Hello, are there any blockers currently preventing the change from landing on
> the stable branch?  If so, do you have any tips on where one can locate them?
>

brlin-tw commented:
@nteodosio 

Thanks for the information!

nteodosio committed:
Fix wrong test command's syntax.

Commit:
Bump version to the latest release candidate (143.0.4-1).

nteodosio committed:
Disable build parallelism in ARMHF.

nteodosio committed:
New upstream release 144.0-1.

nteodosio committed:
LLVM 17.

Commit:
Bug 1994920 - Use langpack-LOCALE instead of langpack-LOCALE.xpi

kabinbr1 commented:
without this firefox uses llvmpipe instead of radeonsi and makes opengl performance in a RX 9060 completely awful

Commit:
Add timezone-control plug

See LP: #2127943
Auto-connection request: https://forum.snapcraft.io/t/firefox-auto-connection-request-for-timezone-control/49007

Silences,

= AppArmor =
Time: 2025-10-15T19:0
Log: apparmor="DENIED" operation="dbus_method_call" bus="system" path="/org/freedesktop/timedate1" interface="org.freedesktop.DBus.Properties" member="GetAll" mask="send" name=":1.302" pid=19390 label="snap.firefox.firefox" peer_pid=19494 peer_label="unconfined"

nteodosio committed:
New upstream release 144.0-2.

nteodosio committed:
Upstream release 145.0-1.

nteodosio committed:
Increase number of jobs for ARMHF builds.

aleasto commented on snapcraft.yaml:
```diff
@@ -452,18 +468,18 @@ parts:
       - libvulkan1
-      - libxt6
+      - libxt6t64
       - mesa-vulkan-drivers
```
Drop: these are provided by `mesa-2404:gpu-2404`

Also drop:
```
      # update the path to the vulkan drivers for the snap environment
      sed -i 's#/usr/lib/#/snap/firefox/current/usr/lib/#' $CRAFT_PART_INSTALL/usr/share/vulkan/icd.d/*.json
```
from `override-build`

aleasto commented on snapcraft.yaml:
```diff
@@ -505,7 +521,7 @@ parts:
       - usr/share/pipewire
       - usr/share/vulkan
```
Drop together with `mesa-vulkan-drivers`

aleasto commented on snapcraft.yaml:
```diff
@@ -505,7 +521,7 @@ parts:
       - usr/share/pipewire
       - usr/share/vulkan
       # Workaround for LP: #2016358 (see the 'override-stage' above).
-      - gnome-platform
+      - gpu-2404
       - data-dir/icons
       - data-dir/sounds
       - data-dir/themes
```
Drop all of this

aleasto commented on snapcraft.yaml:
```diff
@@ -452,18 +468,18 @@ parts:
       - libpipewire-0.3-modules
       - libspa-0.2-modules
       - libspeechd2
       - libvulkan1
```
Drop: this is provided by `mesa-2404:gpu-2404`

aleasto commented on snapcraft.yaml:
```diff
@@ -452,18 +468,18 @@ parts:
       # content interface, while a proper fix is implemented in snapd.
       # Thanks to James Henstridge.
-      mkdir $CRAFT_PART_INSTALL/{gnome-platform,data-dir,data-dir/{icons,sounds,themes}}
+      mkdir $CRAFT_PART_INSTALL/{gpu-2404,data-dir,data-dir/{icons,sounds,themes}}
```
`gnome-platform` is still a used mountpoint, so removing it doesn't make much sense to me, unless the mentioned snapd bug is already fixed.

Since practice shows that snapd is able to mount the `gnome-46-2404` contents in `$SNAP/gnome-platform` even without it being listed here, I believe we can drop this workaround entirely.   _(unless adding `gnome-platform` again here would help with [bug 2043993](https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/2016358)?)_

aleasto commented on snapcraft.yaml:
```diff
@@ -553,24 +569,36 @@ parts:
     stage-packages:
-      - libavcodec58
+      - libavcodec60
+      - libhwy1t64
+      - libsvtav1enc1d1
+      - libjxl0.7
+      - librav1e0
+      # the package is intel specific
+      - on amd64:
+        - libvpl2
```
All of these are included in `gnome-46-2404`

aleasto commented on snapcraft.yaml:
```diff
@@ -452,18 +468,18 @@ parts:
+      - libasound2t64
+      - libcurl4t64
       - libpci3
-      - libpipewire-0.3-0
+      - libpipewire-0.3-0t64
       - libpipewire-0.3-modules
       - libspa-0.2-modules
```
Drop: These are provided by `gnome-46-2404`

aleasto commented on snapcraft.yaml:
```diff
@@ -452,18 +468,18 @@ parts:
-      - libxt6
+      - libxt6t64
       - mesa-vulkan-drivers
       - pipewire-bin
       - pipewire-pulse
```
Why are these installed?

aleasto commented on snapcraft.yaml:
```diff
@@ -452,18 +468,18 @@ parts:
-      - libasound2
-      - libcurl4
+      - libasound2t64
+      - libcurl4t64
```
Drop: These are provided by `gnome-46-2404`

aleasto commented on snapcraft.yaml:
```diff
@@ -452,18 +468,18 @@ parts:
       - libspeechd2
       - libvulkan1
-      - libxt6
+      - libxt6t64
```
Drop: this is provided by `gnome-46-2404`

nteodosio committed:
New upstream release 145.0-2.

Commit:
Bump version to the latest release candidate (145.0.1-1).

nteodosio committed:
Reapply "Add kerberos-tickets plug."

This reverts commit 673af70f9094ad227925d22b66fce139acd1af59.

Commit:
Bump version to the latest release candidate (145.0.2-1).

nteodosio committed:
New upstream release 146.0-1.

nteodosio committed:
Fix FTBFS due to incorrect test environment determination.

Commit:
Bump version to the latest release candidate (146.0-2).

nteodosio committed:
Drop Armhf.

Commit:
Bump version to the latest release candidate (146.0.1-1).

nteodosio committed:
Major version release 147.0-1.

Commit:
Bump version to the latest release candidate (147.0.1-1).

Commit:
Bump version to the latest release candidate (147.0.2-1).

Commit:
Bump version to the latest release candidate (147.0.3-1).

nteodosio committed:
Rust 1.82->1.85.

https://github.com/rust-random/getrandom/commit/59d52049773a4f974da4818cf58f96edfc9159f1
bumped the MSRV from 1.63 to 1.85, causing the build failure

--->
  Downloaded getrandom v0.4.1
error: failed to compile `cbindgen v0.29.2`, intermediate artifacts can be found at `/tmp/cargo-installiLZpaJ`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
Caused by:
  failed to parse manifest at `/builds/worker/.cargo/registry/src/index.crates.io-6f17d22bba15001f/getrandom-0.4.1/Cargo.toml`
Caused by:
  feature `edition2024` is required
  The package requires the Cargo feature called `edition2024`, but that feature is not stabilized in this version of Cargo (1.82.0 (8f40fc59f 2024-08-21)).
  Consider trying a newer version of Cargo (this may require the nightly release).
  See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#edition-2024 for more information about the status of this feature.
<---

From https://firefoxci.taskcluster-artifacts.net/PzwkkgbTQxm1GS31LfATiA/0/public/logs/live_backing.log

Commit:
Bump version to the latest release candidate (147.0.4-1).

nteodosio committed:
Drop D274753.patch, no longer needed.

nteodosio committed:
New upstream release 148.0-1.

nteodosio committed:
Rust 1.85->1.88.

Commit:
Bump version to the latest release candidate (148.0.2-1).

nteodosio committed:
Release 149.0-1.

Commit:
Bump version to the latest release candidate (149.0.2-1).
